diff --git a/kernel-module-source/utils.mk b/kernel-module-source/utils.mk
index 17421ba9f840..749d09da8ec1 100644
--- a/kernel-module-source/utils.mk
+++ b/kernel-module-source/utils.mk
@@ -569,7 +569,7 @@ define READ_ONLY_OBJECT_FROM_FILE_RULE
  $$(OUTPUTDIR)/$$(notdir $(1)).o: $(1)
	$(at_if_quiet)$$(MKDIR) $$(OUTPUTDIR)
	$(at_if_quiet)cd $$(dir $(1)); \
-	$$(call quiet_cmd_no_at,LD) -r -z noexecstack --format=binary \
+	$$(call quiet_cmd_no_at,LD) -r -z noexecstack -m elf_x86_64 --format=binary \
	    $$(notdir $(1)) -o $$(OUTPUTDIR_ABSOLUTE)/$$(notdir $$@)
	$$(call quiet_cmd,OBJCOPY) \
	    --rename-section .data=.rodata,contents,alloc,load,data,readonly \

diff --git a/nvidia-installer-525.53/utils.mk b/nvidia-installer-525.53/utils.mk
index 17421ba9f840..749d09da8ec1 100644
--- a/nvidia-installer-525.53/utils.mk
+++ b/nvidia-installer-525.53/utils.mk
@@ -569,7 +569,7 @@ define READ_ONLY_OBJECT_FROM_FILE_RULE
  $$(OUTPUTDIR)/$$(notdir $(1)).o: $(1)
	$(at_if_quiet)$$(MKDIR) $$(OUTPUTDIR)
	$(at_if_quiet)cd $$(dir $(1)); \
-	$$(call quiet_cmd_no_at,LD) -r -z noexecstack --format=binary \
+	$$(call quiet_cmd_no_at,LD) -r -z noexecstack -m elf_x86_64 --format=binary \
	    $$(notdir $(1)) -o $$(OUTPUTDIR_ABSOLUTE)/$$(notdir $$@)
	$$(call quiet_cmd,OBJCOPY) \
	    --rename-section .data=.rodata,contents,alloc,load,data,readonly \

diff --git a/nvidia-modprobe/utils.mk b/nvidia-modprobe/utils.mk
index 17421ba9f840..749d09da8ec1 100644
--- a/nvidia-modprobe/utils.mk
+++ b/nvidia-modprobe/utils.mk
@@ -569,7 +569,7 @@ define READ_ONLY_OBJECT_FROM_FILE_RULE
  $$(OUTPUTDIR)/$$(notdir $(1)).o: $(1)
	$(at_if_quiet)$$(MKDIR) $$(OUTPUTDIR)
	$(at_if_quiet)cd $$(dir $(1)); \
-	$$(call quiet_cmd_no_at,LD) -r -z noexecstack --format=binary \
+	$$(call quiet_cmd_no_at,LD) -r -z noexecstack -m elf_x86_64 --format=binary \
	    $$(notdir $(1)) -o $$(OUTPUTDIR_ABSOLUTE)/$$(notdir $$@)
	$$(call quiet_cmd,OBJCOPY) \
	    --rename-section .data=.rodata,contents,alloc,load,data,readonly \

diff --git a/nvidia-persistenced/utils.mk b/nvidia-persistenced/utils.mk
index 17421ba9f840..749d09da8ec1 100644
--- a/nvidia-persistenced/utils.mk
+++ b/nvidia-persistenced/utils.mk
@@ -569,7 +569,7 @@ define READ_ONLY_OBJECT_FROM_FILE_RULE
  $$(OUTPUTDIR)/$$(notdir $(1)).o: $(1)
	$(at_if_quiet)$$(MKDIR) $$(OUTPUTDIR)
	$(at_if_quiet)cd $$(dir $(1)); \
-	$$(call quiet_cmd_no_at,LD) -r -z noexecstack --format=binary \
+	$$(call quiet_cmd_no_at,LD) -r -z noexecstack -m elf_x86_64 --format=binary \
	    $$(notdir $(1)) -o $$(OUTPUTDIR_ABSOLUTE)/$$(notdir $$@)
	$$(call quiet_cmd,OBJCOPY) \
	    --rename-section .data=.rodata,contents,alloc,load,data,readonly \

diff --git a/nvidia-settings/src/libXNVCtrl/utils.mk b/nvidia-settings/src/libXNVCtrl/utils.mk
index 17421ba9f840..749d09da8ec1 100644
--- a/nvidia-settings/src/libXNVCtrl/utils.mk
+++ b/nvidia-settings/src/libXNVCtrl/utils.mk
@@ -569,7 +569,7 @@ define READ_ONLY_OBJECT_FROM_FILE_RULE
  $$(OUTPUTDIR)/$$(notdir $(1)).o: $(1)
	$(at_if_quiet)$$(MKDIR) $$(OUTPUTDIR)
	$(at_if_quiet)cd $$(dir $(1)); \
-	$$(call quiet_cmd_no_at,LD) -r -z noexecstack --format=binary \
+	$$(call quiet_cmd_no_at,LD) -r -z noexecstack -m elf_x86_64 --format=binary \
	    $$(notdir $(1)) -o $$(OUTPUTDIR_ABSOLUTE)/$$(notdir $$@)
	$$(call quiet_cmd,OBJCOPY) \
	    --rename-section .data=.rodata,contents,alloc,load,data,readonly \

diff --git a/nvidia-settings/utils.mk b/nvidia-settings/utils.mk
index 17421ba9f840..749d09da8ec1 100644
--- a/nvidia-settings/utils.mk
+++ b/nvidia-settings/utils.mk
@@ -569,7 +569,7 @@ define READ_ONLY_OBJECT_FROM_FILE_RULE
  $$(OUTPUTDIR)/$$(notdir $(1)).o: $(1)
	$(at_if_quiet)$$(MKDIR) $$(OUTPUTDIR)
	$(at_if_quiet)cd $$(dir $(1)); \
-	$$(call quiet_cmd_no_at,LD) $$(RAW_LDFLAGS) -r -z noexecstack --format=binary \
+	$$(call quiet_cmd_no_at,LD) $$(RAW_LDFLAGS) -r -z noexecstack -m elf_x86_64 --format=binary \
	    $$(notdir $(1)) -o $$(OUTPUTDIR_ABSOLUTE)/$$(notdir $$@)
	$$(call quiet_cmd,OBJCOPY) \
	    --rename-section .data=.rodata,contents,alloc,load,data,readonly \

diff --git a/nvidia-xconfig/utils.mk b/nvidia-xconfig/utils.mk
index 17421ba9f840..749d09da8ec1 100644
--- a/nvidia-xconfig/utils.mk
+++ b/nvidia-xconfig/utils.mk
@@ -569,7 +569,7 @@ define READ_ONLY_OBJECT_FROM_FILE_RULE
  $$(OUTPUTDIR)/$$(notdir $(1)).o: $(1)
	$(at_if_quiet)$$(MKDIR) $$(OUTPUTDIR)
	$(at_if_quiet)cd $$(dir $(1)); \
-	$$(call quiet_cmd_no_at,LD) -r -z noexecstack --format=binary \
+	$$(call quiet_cmd_no_at,LD) -r -z noexecstack -m elf_x86_64 --format=binary \
	    $$(notdir $(1)) -o $$(OUTPUTDIR_ABSOLUTE)/$$(notdir $$@)
	$$(call quiet_cmd,OBJCOPY) \
	    --rename-section .data=.rodata,contents,alloc,load,data,readonly \
